src/module_3/Module3_Milestone1.md

```diff
@@ -0,0 +1,1825 @@
+
+
+```python
+print(report)
+```
+
+                  precision    recall  f1-score   support
+    
+               0       0.99      1.00      0.99    467548
+               1       0.00      0.00      0.00      6408
+    
+        accuracy                           0.99    473956
+       macro avg       0.49      0.50      0.50    473956
+    weighted avg       0.97      0.99      0.98    473956
+    
+
+
+
+```python
+# Generate the confusion matrix
+conf_matrix = confusion_matrix(Y_val, Y_pred)
+
+# Print the confusion matrix
+print("Confusion Matrix:")
+print(conf_matrix)
+```
+
+    Confusion Matrix:
+    [[467548      0]
+     [  6408      0]]
+
+
+It is worth mentioning the confusion matrix, the true negatives are located in the top-left, the false positives in the top-right, the false negatives in the bottom-left, and, finally, in the bottom right, the true positives. 
+
+The first thing we can observe from this report is that there is a clear imbalance between classes: class 0 (negative) represents the majority class, with 467548, whereas class 1 (positive) contains only 6408 instances.
+
+On one hand, for the negative class (0), from the high precision, we can observe that nearly all instances predicted as class 0 are indeed class 0, which suggests very few false positives for class 0. Furthermore, the recall is 100%, indicating that almost all 0 instances are correctly predicted by the model. Finally, a high F1 score for class 0 indicates excellent precision and recall balance. In conclusion, the model is very effective at identifying class 0 instances. All this can also be observed in the first row of the confusion matrix, as the number of true  negatives is 467455 and the number of false positives is only 93 (the model incorrectly predicted 93 cases as positive when they were actually negative).
+
+On the other hand, for the positive class (1), the results are not as good. Its precision of 72% indicates that, of all instances predicted as 1, only 72% actually belonged to class 1. Additionally, the model's recall for this class is extremely low: only 4% of the actual cases with outcome 1  are correctly identified, indicating the model struggles significantly to detect class 1 instances. Finally, as we would have expected, the F1 score is very low, reflecting how poor performance is in terms of both precision and recall. Once again, this can also be seen in the confusion matrix: it has a really low number of true positives (245), and a really high number of false positives (6163).
+
+In conclusion, we can observe that the model is very effective at identifying class 0 intances, but for cases where the outcome is 1 it performs extremely poorly,  indicating the model is biased towards the majority class, as it has been trained on an imbalanced dataset. 
```

Comment:
This conclusions cannot be drawn if we do not compare against a baseline. How much better are we identifying the 0s than the baseline? That is what will tell us if we are effective or not